Build with LTO fails

Build with [LTO](https://wiki.debian.org/LTO), with flags -flto -Werror=lto-type-mismatch produce the following error:

`x86_64-pc-linux-gnu-gcc -I/usr/include/libxml2    -march=native -O2 -pipe -mmmx -msse4.2 -mssse3 -mfpmath=sse -fomit-frame-pointer      -frecord-gcc-switches -flto -Werror=odr -Werror=lto-type-mismatch -Werror=strict-aliasing  -Wl,-O1 -Wl,--as-needed -Wl,--defsym=__gentoo_check_ldflags__=0 -o automatic ../src/automatic.o ../src/base64.o ../src/config_parser.o ../src/downloads.o ../src/feed_item.o ../src/file.o ../src/json.o ../src/list.o ../src/output.o ../src/filters.o ../src/prowl.o ../src/pushover.o ../src/regex.o ../src/rss_feed.o ../src/state.o ../src/torrent.o ../src/transmission.o ../src/urlcode.o ../src/utils.o ../src/web.o ../src/xml_parser.o -lcurl  -lxml2  -lpcre  -lcurl 
../include/downloads.h:30:5: error: type of 'addToBucket' does not match original declaration [-Werror=lto-type-mismatch]
   30 | int addToBucket(const char* identifier, NODE **head, const uint16_t maxBucketItems);
      |     ^
../src/downloads.c:89:5: note: type mismatch in parameter 3
   89 | int addToBucket(const char* guid, NODE **head, const int maxBucketItems) {
      |     ^
../src/downloads.c:89:5: note: type 'int' should match type 'uint16_t'
../src/downloads.c:89:5: note: 'addToBucket' was previously declared here
../src/downloads.c:89:5: note: code may be misoptimized unless '-fno-strict-aliasing' is used
../src/memwatch.c: In function 'mwARI':
../src/memwatch.c:836:12: warning: ignoring return value of 'fgets' declared with attribute 'warn_unused_result' [-Wunused-result]
  836 |     (void) fgets(inbuf,sizeof(inbuf),stdin);
      |            ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
lto1: some warnings being treated as errors
lto-wrapper: fatal error: x86_64-pc-linux-gnu-gcc returned 1 exit status
compilation terminated.
/usr/lib/gcc/x86_64-pc-linux-gnu/11.3.0/../../../../x86_64-pc-linux-gnu/bin/ld: error: lto-wrapper failed
collect2: error: ld returned 1 exit status
make[3]: *** [Makefile:668: automatic] Error 1
make[3]: *** Waiting for unfinished jobs....
make[3]: Leaving directory '/var/tmp/portage/net-p2p/automatic-0.9.0/work/Automatic-0.9.0/src'
make[2]: *** [Makefile:1149: all-recursive] Error 1
make[2]: Leaving directory '/var/tmp/portage/net-p2p/automatic-0.9.0/work/Automatic-0.9.0/src'
make[1]: *** [Makefile:518: all] Error 2
make[1]: Leaving directory '/var/tmp/portage/net-p2p/automatic-0.9.0/work/Automatic-0.9.0/src'
make: *** [Makefile:369: all-recursive] Error 1
`

--- a/src/automatic.c
+++ b/src/automatic.c
@@ -729,7 +729,7 @@ int main(int argc, char **argv) {
     config_file = am_strdup(AM_DEFAULT_CONFIGFILE);
   }
 
-  strncpy(AutoConfigFile, config_file, strlen(config_file));
+  strcpy(AutoConfigFile, config_file);
 
   ses = session_init();
   ses->match_only = match_only;

--- a/src/downloads.c
+++ b/src/downloads.c
@@ -86,9 +86,9 @@ uint8_t has_been_downloaded(const simple_list bucket, const feed_item item) {
  * The size of the provided bucket list is kept to maxBucketItems.
  * If it gets larger than the specified value, the oldest element is removed from the list.
  */
-int addToBucket(const char* guid, NODE **head, const int maxBucketItems) {
+int addToBucket(const char* guid, NODE **head, const uint16_t maxBucketItems) {
 	addToHead(am_strdup(guid), head);
-	if(maxBucketItems > 0 && listCount(*head) > (uint32_t)maxBucketItems) {
+	if(maxBucketItems > 0 && listCount(*head) > maxBucketItems) {
 		dbg_printf(P_INFO2, "[add_to_bucket] bucket gets too large, deleting last item...\n");
 		removeLast(*head, NULL);
 	}

--- a/src/filters.c
+++ b/src/filters.c
@@ -85,6 +85,8 @@ PUBLIC void filter_printList(simple_list list) {
 		cur = cur->next;
 	}
 	dbg_printf(P_INFO2, "------- end  -------------\n");
+#else
+	(void)list;
 #endif
 }
 

--- a/src/memwatch.c
+++ b/src/memwatch.c
@@ -1749,7 +1749,7 @@ static int mwRelink( mwData* mw, const char* file, int line ) {
 			if( mw1->next )
 			{
 				if( !mwIsReadAddr(mw1->next,mwDataSize) ||
-					!mw1->next->check != CHKVAL(mw1) ||
+					(!mw1->next->check) != CHKVAL(mw1) ||
 					mw1->next->prev != mw1 )
 				{
 					mwWrite("relink: forward chain's last intact MW is MW-%p, %ld %sbytes at %s(%d)\n",
@@ -1791,7 +1791,7 @@ scan_reverse:
             if( mw2->prev )
 			{
 				if( !mwIsReadAddr(mw2->prev,mwDataSize) ||
-					!mw2->prev->check != CHKVAL(mw2) ||
+					(!mw2->prev->check) != CHKVAL(mw2) ||
 					mw2->prev->next != mw2 )
 				{
 					mwWrite("relink: reverse chain's last intact MW is MW-%p, %ld %sbytes at %s(%d)\n",

--- a/src/rss_feed.c
+++ b/src/rss_feed.c
@@ -90,6 +90,8 @@ void feed_printList(simple_list list) {
       cur = cur->next;
    }
    dbg_printf(P_INFO2, "------- end  -------------\n");
+#else
+   (void)list;
 #endif
 }
 

--- a/src/torrent.c
+++ b/src/torrent.c
@@ -53,7 +53,7 @@ get_filename(char *path, const char *content_filename, const char* url, const ch
 
   if (content_filename) {
     dbg_printf(P_INFO, "Content-Filename: %s", content_filename);
-    strncpy(buf, content_filename, strlen(content_filename) + 1);
+    strcpy(buf, content_filename);
   } else {
     strcpy(tmp, url);
     p = strtok(tmp, "/");

